shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -967,7 +986,22 @@ - (void)onAccessibilityStatusChanged:(NSNotification*)notification {
 
   self.semanticsEnabled = enabled;
 }
+- (void)handleAccessibilityEvent:(NSDictionary<NSString*, id>*)annotatedEvent {
+  NSString* type = annotatedEvent[@"type"];
+  if ([type isEqualToString:@"announce"]) {
+    NSString* message = annotatedEvent[@"data"][@"message"];
+    NSNumber* assertiveness = annotatedEvent[@"data"][@"assertiveness"];
+
+    NSAccessibilityPriorityLevel priority = [assertiveness isEqualToNumber:@1]
+                                                ? NSAccessibilityPriorityHigh
+                                                : NSAccessibilityPriorityMedium;
 
+    NSAccessibilityPostNotificationWithUserInfo(
+        [self viewControllerForId:kFlutterDefaultViewId].flutterView,
+        NSAccessibilityAnnouncementRequestedNotification,
+        @{NSAccessibilityAnnouncementKey : message, NSAccessibilityPriorityKey : @(priority)});
+  }
```

Comment:
Top-level AppKit API functions are always a bit tricky to test. There are a few tricks you can use for this. A simple one in this case is:
1. Extract this to a method like:
```objc
- (void)announceAccessibilityMessage:(NSString*)withPriority:(NSAccessibilityPrioirityKey);
```
2. Use `CreateMockFlutterEngine()` to create a partial mock in your test.
3. Verify the method got called in the test:
```objc
[[engineMock expect] announceAccessibilityMessage:@"Foo" withPriority:...];

// Fire an event here.

[engineMock verify];
```